modules/database/src/ioc/db/dbChannel.c

```diff
@@ -630,14 +632,61 @@ long dbChannelOpen(dbChannel *chan)
 }
 
 /* Only use dbChannelGet() if the record is already locked. */
-long dbChannelGet(dbChannel *chan, short type, void *pbuffer,
-        long *options, long *nRequest, void *pfl)
+long dbChannelGet(dbChannel *chan, short dbrType, void *pbuffer,
+        long *options, long *nRequest, db_field_log *pfl)
 {
-    return dbGet(&chan->addr, type, pbuffer, options, nRequest, pfl);
+    dbAddr addr = chan->addr; /* structure copy */
+
+    if (pfl) {
+        addr.field_size = pfl->field_size;
+        addr.field_type = pfl->field_type;
+        addr.no_elements = pfl->no_elements;
+        addr.pfield = dbfl_pfield(pfl);
+    }
+
+    /*
+     * Try to optimize scalar requests by caching (just) the conversion
+     * routine. This is possible only if we have no options, the field and the
+     * request have exactly one element, we have no special processing to do,
+     * and the field type is not DBF_DEVICE or larger.
+     * Note that if the db_field_log has already copied the data, dbGet
+     * does not call get_array_info.
+     */
+    if (!(options && *options)
+        && addr.no_elements == 1
+        && (!nRequest || *nRequest == 1)
+        && (dbfl_has_copy(pfl) || addr.special != SPC_DBADDR)
+        && addr.special != SPC_ATTRIBUTE
+        && addr.field_type <= DBF_DEVICE)
+    {
+        if (chan->getCvt && chan->lastGetdbrType == dbrType) {
+            return chan->getCvt(addr.pfield, pbuffer, &addr);
+        } else {
+            unsigned short dbfType = addr.field_type;
+
+            if (INVALID_DB_REQ(dbrType)) {
+                char message[80];
+
+                sprintf(message, "dbChannelGet: Request type is %d\n", dbrType);
+                recGblDbaddrError(S_db_badDbrtype, &addr, message);
+                return S_db_badDbrtype;
+            }
+
+            chan->getCvt = dbFastGetConvertRoutine[dbfType][dbrType];
+            chan->lastGetdbrType = dbrType;
+            if (nRequest)
+                *nRequest = addr.no_elements;
```

Comment:
I'm surprised to see that `nRequest` is only updated on the "slow" path.  I would expect this to be done by both or neither.

Reply:
What's the point if we already know that
```
        (!nRequest || *nRequest == 1)
```
?